### PR TITLE
add support for STORAGE table option

### PR DESCRIPTION
**Pull request by leiysky**
### What problem does this PR solve? 
Add support for STORAGE table option.

Issue: #403 

MySQL syntax:
```
table_option:
  | TABLESPACE tablespace_name 
  | STORAGE {DISK|MEMORY}
...
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by leoppro** [comment]
thanks for your contribution!

sign the agreement please @leiysky

File: parser.y [hunk 1]
```diff
@@ -7627,6 +7629,16 @@ TableOption:
...
 	}
+|	"STORAGE" "MEMORY"
+	{
+		// Parse it but will ignore it.
+		$$ = &ast.TableOption{Tp: ast.TableOptionStorageMedia, StrValue: "MEMORY"}
+	}
+|	"STORAGE" "DISK"
+	{
+		// Parse it but will ignore it.
+		$$ = &ast.TableOption{Tp: ast.TableOptionStorageMedia, StrValue: "DISK"}
```
Comment (leoppro) [inline parser.y]: throw a warn like this:
```
		yylex.AppendError(yylex.Errorf("The STORAGE clause is parsed but ignored by all storage engines."))
		parser.lastErrorAsWarn()
```

File: ast/ddl.go [hunk 2]
```diff
@@ -1618,6 +1619,11 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
 		ctx.WriteKeyWord("INDEX DIRECTORY ")
 		ctx.WritePlain("= ")
 		ctx.WriteString(n.StrValue)
+	case TableOptionStorageMedia:
+		// TODO: not support
+		ctx.WriteKeyWord("STORAGE ")
+		ctx.WriteKeyWord(n.StrValue)
+		ctx.WritePlain(" /* TableOptionStorageMedia is not supported */ ")
```
Comment (leoppro) [inline ast/ddl.go]:
```suggestion
```

File: ast/ddl.go [hunk 3]
```diff
@@ -1618,6 +1619,11 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
 		ctx.WriteKeyWord("INDEX DIRECTORY ")
 		ctx.WritePlain("= ")
 		ctx.WriteString(n.StrValue)
+	case TableOptionStorageMedia:
+		// TODO: not support
```
Comment (leoppro) [inline ast/ddl.go]:
```suggestion
```

**Comment by leoppro** [comment]
@kennytm  PTAL

File: parser_test.go [hunk 4]
```diff
@@ -1735,6 +1735,8 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"create table t (c int) PACK_KEYS = 1", true, "CREATE TABLE `t` (`c` INT) PACK_KEYS = DEFAULT /* TableOptionPackKeys is not supported */ "},
 		{"create table t (c int) PACK_KEYS = 0", true, "CREATE TABLE `t` (`c` INT) PACK_KEYS = DEFAULT /* TableOptionPackKeys is not supported */ "},
 		{"create table t (c int) PACK_KEYS = DEFAULT", true, "CREATE TABLE `t` (`c` INT) PACK_KEYS = DEFAULT /* TableOptionPackKeys is not supported */ "},
+		{"create table t (c int) STORAGE DISK", true, "CREATE TABLE `t` (`c` INT) STORAGE DISK /* TableOptionStorageMedia is not supported */ "},
```
Comment (leiysky) [inline parser_test.go]:
```suggestion
		{"create table t (c int) STORAGE DISK", true, "CREATE TABLE `t` (`c` INT) STORAGE DISK"},
```

File: parser_test.go [hunk 5]
```diff
@@ -1735,6 +1735,8 @@ func (s *testParserSuite) TestDDL(c *C) {
...
+		{"create table t (c int) STORAGE MEMORY", true, "CREATE TABLE `t` (`c` INT) STORAGE MEMORY /* TableOptionStorageMedia is not supported */ "},
```
Comment (leiysky) [inline parser_test.go]: {"create table t (c int) STORAGE MEMORY", true, "CREATE TABLE `t` (`c` INT) STORAGE MEMORY"},

**Comment by leiysky** [comment]
@leoppro Resolved

**Comment by kennytm** [comment]
@leiysky Thanks for the fix, but the [CLA still needs to be signed](https://github.com/pingcap/parser/pull/407#issuecomment-515932877), otherwise we cannot merge the PR.

**Pull request merged**